components/proxy/src/main/java/com/hotels/styx/routing/handlers/HostProxy.java

```diff
@@ -303,22 +306,19 @@ public static HostProxy createHostProxyHandler(
                     .id(originId)
                     .build();
 
-            OriginMetrics originMetrics = new OriginMetrics(meterRegistry, origin.id().toString(), origin.applicationId().toString());
-
-            ConnectionPool.Factory connectionPoolFactory = new SimpleConnectionPoolFactory.Builder()
-                    .connectionFactory(
-                            connectionFactory(
-                                    executor,
-                                    tlsSettings,
-                                    responseTimeoutMillis,
-                                    maxHeaderSize,
-                                    theOrigin -> originMetrics,
-                                    poolSettings.connectionExpirationSeconds()))
-                    .connectionPoolSettings(poolSettings)
-                    .meterRegistry(meterRegistry)
-                    .build();
+            Tags tags = Tags.of("host", host).and("port", valueOf(port));
+            RequestMetrics requestMetrics = new RequestMetrics(meterRegistry, tags);
 
-            return new HostProxy(host, port, StyxHostHttpClient.create(connectionPoolFactory.create(origin)), originMetrics);
+            Connection.Factory connectionFactory = connectionFactory(
+                    executor,
+                    tlsSettings,
+                    responseTimeoutMillis,
+                    maxHeaderSize,
+                    o -> requestMetrics,
+                    poolSettings.connectionExpirationSeconds());
+            ConnectionPool connectionPool = new SimpleConnectionPool(origin, poolSettings, connectionFactory);
+            connectionPool = new StatsReportingConnectionPool(connectionPool, meterRegistry, tags);
+            return new HostProxy(host, port, StyxHostHttpClient.create(connectionPool), requestMetrics);
```

Comment:
This looks like a decent change. 👍 

However, in order to keep the coding style consistent, I would recommend inlining some unnecessary stack variables. Such as `connectionFactory` and `connectionPool`. 

```
return new HostProxy(host, port, StyxHostHttpClient.create(
    new StatsReportingConnectionPool(
        new SimpleConnectionPool(
            origin, 
            poolSettings, 
            connectionFactory(
                     executor,
                     tlsSettings,
                     responseTimeoutMillis,
                     maxHeaderSize,
                     o -> requestMetrics,
                     poolSettings.connectionExpirationSeconds())), 
        meterRegistry, 
        tags)
    ), 
    requestMetrics);
```

This reduces cognitive load because two stack variables and three assignments are eliminated. I think the other routing objects follow similar style.

Reply:
Also well spotted to remove the pool factory. 👍